server/middlewares/validators/videos/video-source.ts

```diff
@@ -0,0 +1,44 @@
+import { getVideoWithAttributes } from '@server/helpers/video'
+import { MVideoFullLight } from '@server/types/models'
+import { HttpStatusCode, UserRight, VideoPrivacy } from '@shared/models'
+import express from 'express'
+import { logger } from '../../../helpers/logger'
+import {
+  areValidationErrors,
+  doesVideoExist,
+  isValidVideoIdParam
+} from '../shared'
+
+const videoSourceGetValidator = [
+  isValidVideoIdParam('id'),
+
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    logger.debug('Checking videoSourceGet parameters', { parameters: req.params })
+
+    if (areValidationErrors(req, res)) return
+    if (!await doesVideoExist(req.params.id, res, 'for-api')) return
```

Comment:
Hum I think you're right. I assumed that we should use the SEE right, since we just see the video filename. But maybe the UPDATE right is more appropriate.

Then you can use `checkUserCanManageVideo` function